ui/static/ui/js/learning_resources.jsx

```diff
@@ -120,8 +127,65 @@ define('learning_resources', [
         _.filter(e.target.options, function(option) {
           return option.selected && option.value !== null;
         }), 'value');
-
-      this.props.setValues(this.props.selectedVocabulary.slug, selectedValues);
+      // check if the current vocabulary allows free tagging and in case add
+      // the new tags before proceeding
+      if (this.props.selectedVocabulary.vocabulary_type === 'f') {
+        // extract the list of terms not in the vocabulary existing terms
+        var termsSlugs = _.pluck(this.props.selectedVocabulary.terms, 'slug');
+        var termsToCreate = _.filter(selectedValues, function (slug) {
+          return _.indexOf(termsSlugs, slug) === -1;
+        });
+        // if there are no new terms to create, set the state
+        if (!termsToCreate.length) {
+          this.props.setValues(
+            this.props.selectedVocabulary.slug, selectedValues
+          );
+        } else {
```

Comment:
I'm not sure I understood: If there is one term added and another removed in the same event (and I don't think this can even happen) it means that the selected terms will include the one added and not the one removed and everything else will work just fine, even if the added term is a new term.